.github/workflows/sdk-size-updates.yml

```diff
@@ -0,0 +1,97 @@
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-java
+      - uses: ./.github/actions/gradle-cache
+        with:
+          key-prefix: gradle-build
+
+      - name: Assemble SDKs
+        run: |
+          for module in $MODULES; do
+            modules+=" :metrics:$module-metrics-baseline:assemble :metrics:$module-metrics-stream:assemble"
+          done
+          ./gradlew $modules
+
+      - name: Update size metrics
+        run: |
+          # Create temporary JSON file
+          echo '{}' > metrics.json
+
+          # Calculate sizes
+          for module in $MODULES; do
+            for variant in $VARIANTS; do
+              baselineFile="metrics/$module-metrics-baseline/build/outputs/apk/$variant/$module-metrics-baseline-$variant.apk"
+              streamFile="metrics/$module-metrics-stream/build/outputs/apk/$variant/$module-metrics-stream-$variant.apk"
+
+              # Ensure files exist
+              if [[ -f "$baselineFile" && -f "$streamFile" ]]; then
+                baselineSize=$(du -k "$baselineFile" | awk '{print $1}')
+                streamSize=$(du -k "$streamFile" | awk '{print $1}')
+                size=$((streamSize - baselineSize))
+              else
+                echo "Warning: $baselineFile or $streamFile not found. Setting size to 0."
+                size=0
+              fi
+          
+              # Update JSON
+              jq --arg module "$module" --arg variant "$variant" --argjson size "$size" \
+                ".\"$variant\".\"$module\" = $size" metrics.json > temp.json && mv temp.json metrics.json
+            done
+          done
+
+          # Validate Generated JSON
+          jq . metrics.json
+          
+          # Move temporary JSON file to the final file
+          mv metrics.json $METRICS_FILE
+
+      - name: Update size badges
+        run: |
+          for module in $MODULES; do
+            size=$(jq --arg module "$module" ".release.\"$module\"" $METRICS_FILE)
+            sizeInMb=$(echo "scale=2; $size / 1024" | bc)
+            badgeUrl="https://img.shields.io/badge/${module//-/--}-$sizeInMb%20MB-lightgreen"
+            sed -i "s|!\[$module\](.*)|![$module](${badgeUrl})|" README.md
+          done
+
+      - name: Commit and Push JSON
+        run: |
+          git fetch origin $BRANCH_NAME
+          git checkout $BRANCH_NAME
+          
+          git config user.name "github-actions[bot]"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"         
+
+          # Add and commit updated metrics file
+          git add $METRICS_FILE
+          git commit -m "Update SDK size metrics" || echo "No metrics changes to commit"
+          
+          # Add and commit updated README file
+          git add README.md
+          git commit -m "Update SDK size badges" || echo "No README changes to commit"
+
+          git push origin HEAD:$BRANCH_NAME
```

Comment:
`develop` branch is protected, and any commit needs to be pushed using a PR. The only allowed "user" is a bot.
On this [GH Action](https://github.com/GetStream/stream-chat-android/blob/develop/.github/workflows/release-post.yaml#L30-L56) you can see how we use is

Reply:
The script is already using the GitHub action bot 🤔

Reply:
No, it isen.
It is only configuring git properties, but it is not using the bot credentials

Reply:
How come? 🤔 
<img width="531" alt="Screenshot 2025-01-02 at 12 00 28" src="https://github.com/user-attachments/assets/c0774ac0-9072-4ae8-a7a3-708036ce30ec" />

Reply:
<img width="463" alt="Screenshot 2025-01-02 at 12 06 53" src="https://github.com/user-attachments/assets/0d2f946c-b15f-4e0e-ac80-d6b0e1328a5e" />

Reply:
It is because it shows the git properties for the commit itself, but they are not the credentials.
You could configure them within your local machine as well